build: various improvements

tiagolobocastro opened the pull request:
ci(gha/submodule): ensure submodules point to the right branch
    
    Will help avoid mistakes if anyone uses update --remote...
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    chore: don't show crate version on log
    
    We never update the crate version, and it's actually very misleading...
    So let's hide it until we start to update it properly.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    build(images): add tag argument
    
    Allows one to explicitly build the binaries with a given tag, as opposed to the alias-tag
    which simply adds a docker image tag alias.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
build(images): add tag argument

Allows one to explicitly build the binaries with a given tag, as opposed to the alias-tag
which simply adds a docker image tag alias.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
chore: don't show crate version on log

We never update the crate version, and it's actually very misleading...
So let's hide it until we start to update it properly.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro committed:
ci(gha/submodule): ensure submodules point to the right branch

Will help avoid mistakes if anyone uses update --remote...

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

niladrih commented on scripts/git/set-submodule-branches.sh:
```diff
@@ -0,0 +1,55 @@
+
+
+
+if [ "$BRANCH" == "develop" ]; then
```
Could both of these be inside the same if condition. They both do the same thing `SET_BRANCH="${BRANCH}"`

tiagolobocastro replied:
Yes indeed

niladrih reviewed (commented):
minor comment

tiagolobocastro commented:
bors merge

tiagolobocastro commented:
bors merge

tiagolobocastro commented:
hmm strange errors from the bdd tests...
bors merge

tiagolobocastro committed:
chore: pin requests until fixed upstream

There's a compat issue between urllib3 and requests which causes the docker usage in the
pytests to fail.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>
